src/PlanTrip/Trip.tsx

```diff
@@ -0,0 +1,141 @@
+import { Feature, Leg, TripPattern } from "../types";
+import { ActionPanel, Action, Color, List } from "@raycast/api";
+import { fetchTrip } from "../api";
+import { useCallback, useEffect, useState } from "react";
+import {
+  getTransportIcon,
+  formatAsClock,
+  formatTimeDifferenceAsClock,
+  formatMetersToHuman,
+} from "../utils";
+import Accessory = List.Item.Accessory;
+
+const buildAccessories = (trip: TripPattern, isDetailsVisible: boolean): Accessory[] => {
+  if (isDetailsVisible) {
+    return [
+      {
+        tag: {
+          value: formatTimeDifferenceAsClock(trip.expectedStartTime, trip.expectedEndTime),
+          color: Color.Blue,
+        },
+      },
+    ];
+  }
+  const accessories = trip.legs
+    .filter((leg) => !(leg.mode === "foot" && leg.fromPlace.name === leg.toPlace.name))
+    .map((leg) => ({
+      icon: getTransportIcon(leg.mode),
+      text: `${leg.line ? leg.line?.publicCode : ""}`,
```

Comment:
Can simplify this a bit, which also fixes a bug for public code `null`

```suggestion
      text: leg.line?.publicCode,
```

Reply:
I was probably drunk when I wrote this line